src/Simplex/Chat/Store.hs

```diff
@@ -281,12 +310,233 @@ getConnectionChatDirection st userId agentConnId =
           (userId, contactId)
     toContact contactId c [(localContactRef, contactRef, displayName)] =
       let profile = Profile {contactRef, displayName}
-       in Right Contact {contactId, localContactRef, profile, activeConn = c}
+       in Right $ CContact Contact {contactId, localContactRef, profile, activeConn = c}
     toContact _ _ _ = Left $ SEInternal "referenced contact not found"
 
+-- | creates completely new group with a single member - the current user
+createNewGroup :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> TVar ChaChaDRG -> User -> GroupProfile -> m Group
+createNewGroup st gVar User {userId, userContactId, profile} p@GroupProfile {groupRef, displayName} =
+  liftIOEither . checkConstraint SEDuplicateGroupRef . withTransaction st $ \db -> do
+    -- group inserted before profile to ensure its local_group_ref is unique
+    DB.execute db "INSERT INTO groups (local_group_ref, lgr_base, user_id) VALUES (?, ?, ?)" (groupRef, groupRef, userId)
```

Comment:
for `createNewGroup` there's no lgr suffix increment logic because it's up to user to retry, right?

Reply:
yes